meilisearch/index.py

```diff
@@ -153,3 +143,650 @@ def get_index(config, uid):
+                self.config.paths.index,
+                self.uid,
+                self.config.paths.update
+            )
+        )
+
+    def get_update_status(self, update_id):
+        """Get one update from MeiliSearch
+
+        Parameters
+        ----------
+        update_id: int
+            identifier of the update to retrieve
+        Returns
+        ----------
+        update: `list`
+            List of all enqueued and processed actions of the index.
+        """
+        return self.http.get(
+            '{}/{}/{}/{}'.format(
+                self.config.paths.index,
+                self.uid,
+                self.config.paths.update,
+                update_id
+            )
+        )
+
+    def get_stats(self):
+        """Get stats of an index
+
+        Get information about the number of documents, field frequencies, ...
+        https://docs.meilisearch.com/references/stats.html
+        Returns
+        ----------
+        stats: `dict`
+            Dictionnary containing stats about the given index.
+        """
+        return self.http.get(
+            '{}/{}/{}'.format(
+                self.config.paths.index,
+                self.uid,
+                self.config.paths.stat,
+            )
+        )
+
+    # pylint: disable=dangerous-default-value
+    # Not dangerous because opt_params is not modified in the method
+    # See: https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python
+    def search(self, query, opt_params={}):
+        """Search in meilisearch
+
+        Parameters
+        ----------
+        query: str
+            String containing the searched word(s)
+        opt_params: dict
+            Dictionnary containing optional query parameters
+            https://docs.meilisearch.com/references/search.html#search-in-an-index
+        Returns
+        ----------
+        results: `dict`
+            Dictionnary with hits, offset, limit, processingTime and initial query
+        """
+        search_param = {'q': query}
+        params = {**search_param, **opt_params}
+        return self.http.get(
+            '{}/{}/{}?{}'.format(
+                self.config.paths.index,
+                self.uid,
+                self.config.paths.search,
+                urllib.parse.urlencode(params))
+        )
+
+    def get_document(self, document_id):
+        """Get one document with given document identifier
+
+        Parameters
+        ----------
+        document_id: str
+            Unique identifier of the document.
+        Returns
+        ----------
+        document: `dict`
+            Dictionnary containing the documents information
+        """
+        return self.http.get(
+            '{}/{}/{}/{}'.format(
+                self.config.paths.index,
+                self.uid,
+                self.config.paths.document,
+                document_id
+            )
+        )
+
+    def get_documents(self, parameters=None):
```

Comment:
Is this not considered as a default value?
or is 
`  # pylint: disable=dangerous-default-value`

Also disabled for this line ?

Reply:
`None` is not considered a dangerous default value:

https://satran.in/b/python--dangerous-default-value-as-argument